### PR TITLE
Adding ready to run Jupyter notebooks for various ML frameworks

### PR DESCRIPTION
**Pull request by jacekwachowiak**
As by #704, I edited and formatted my notebooks, which I used to check Clipper capabilities. For now only Docker version, when I have a chance I will try to adapt my K8s versions as well. `TensorFlow` is the most extensive one, then `PyTorch`, all others follow the same scheme but may lack comments on each step.

### CODE REVIEW

**Comment by withsmilo** [comment]
WOW!~~~~~ Great!

**Comment by withsmilo** [comment]
@simon-mo @rkooo567 We have to check to run these notebooks with the latest Clipper before merging.

**Comment by rkooo567** [comment]
@jacekwachowiak This is really amazing!! Thanks a lot for the contribution! 

@withsmilo Yeah I agree! Once the next version is released, let's verify if there's anything broken and merge it.

**Comment by rkooo567** [comment]
jenkins okay to test

**Comment by jacekwachowiak** [comment]
The things worth noticing - I used custom Docker images for `Scikit` and `XGBoost` since I couldn't add the packages directly (proxy), and then replaced the argument for upload so I hope there are no typos. And for `MXNet` the version I used was 1.1.0.

**Comment by rkooo567** [comment]
@simon-mo my bad about jenkins test. What's the command for stopping jenkins okay to test?

**Comment by rkooo567** [comment]
@jacekwachowiak. This will be merged after the new version is released!

**Comment by withsmilo** [comment]
@jacekwachowiak May I ask you to write a new Jupyter notebook in your style that trains and serves `Keras` models? The link below is a Jupyter notebook I wrote for a sample.
> https://github.com/withsmilo/How-to-serve-pretrained-models-using-Clipper/blob/master/sample/keras_on_docker_example.ipynb

**Comment by withsmilo** [comment]
@rkooo567 @jacekwachowiak @simon-mo 
I'm checking these Jupyter notebooks using Clipper v0.4.1.
- [x] mxnet_on_docker_example.ipynb
  * mxnet version on local is 1.4.1.
  * mxnet version on container is 1.4.1.
- [x] pyspark_on_docker_example.ipynb
  * pyspark version on local is 2.4.3.
  * pyspark version on container is 2.3.3. (need to update?)
- [x] pytorch_on_docker_example.ipynb
  * torch version on local is 1.1.0.
  * torch version on container is 1.0.1.post2. (will be updated. https://github.com/ucbrise/clipper/pull/726)
- [x] tensorflow_on_docker_example.ipynb
  * tensorflow version on local is 1.13.1.
  * tensorflow version on container is 1.13.1.
- [x] scikit_on_docker_example.ipynb
  * scikit-learn version on local is 0.21.2.
- [x] xgboost_on_docker_example.ipynb
  * xgboost version on local is 0.90.

File: examples/frameworks/mxnet_on_docker_example.ipynb [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+       "'1.1.0'"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "mx.__version__ # seems like the latest version is a bit unstable, so I use 1.1.0"
```
Comment (withsmilo) [inline examples/frameworks/mxnet_on_docker_example.ipynb]: I had successful test with mxnet 1.4.1. So please modify your version in this notebook.

Comment (jacekwachowiak) [reply examples/frameworks/mxnet_on_docker_example.ipynb]: Done!

**Comment by jacekwachowiak** [comment]
> @jacekwachowiak May I ask you to write a new Jupyter notebook in your style that trains and serves `Keras` models? The link below is a Jupyter notebook I wrote for a sample.
> 
> > https://github.com/withsmilo/How-to-serve-pretrained-models-using-Clipper/blob/master/sample/keras_on_docker_example.ipynb

Sure I can prepare something similar to the others, but the image example is imho perfectly fine too! 
In fact I think you should add it to the repo as `image query` example, since it is more understandable than the existing image notebook.

File: examples/frameworks/mxnet_on_docker_example.ipynb (hunk 1 above)
Comment (withsmilo) [reply examples/frameworks/mxnet_on_docker_example.ipynb]: Thanks! :satisfied:

**Comment by withsmilo** [comment]
@jacekwachowiak I agree with you, and will push my examples to this PR as soon as possible.

**Comment by jacekwachowiak** [comment]
Two things to note:
1 - the example uses a text file, the same dataset as in Scikit but it's not included by default in Keras library. If that's not ok, then the model will have to be changed
2 - running `.stop_models` fails, so it's commented out, I'll check a bit more, and if necessary open an issue EDIT: #729 EDIT2: fixed for Docker, notebook ready.

**Review by withsmilo — APPROVED** [review]
LGTM

**Comment by withsmilo** [comment]
@jacekwachowiak 
I added my notebooks to `examples/image_query` directory. Thanks for your great works again!

@simon-mo @rkooo567
All notebooks are ready now. Please review. :blush:

**Comment by simon-mo** [comment]
@jacekwachowiak @withsmilo Thanks for the great works. This is amazing.

@rkooo567 can you make sure these notebooks are referenced in the website?

**Comment by rkooo567** [comment]
@simon-mo I am a little busy this weekday. I will for sure handle all of them this Sat.

**Pull request merged**